src/test/lib/storage/variable_string_dictionary_segment_test.cpp

```diff
@@ -0,0 +1,230 @@
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base_test.hpp"
+
+#include "storage/chunk_encoder.hpp"
+#include "storage/segment_encoding_utils.hpp"
+#include "storage/value_segment.hpp"
+#include "storage/variable_string_dictionary_segment.hpp"
+#include "storage/vector_compression/fixed_width_integer/fixed_width_integer_vector.hpp"
+
+namespace hyrise {
+
+class StorageVariableStringDictionarySegmentTest : public BaseTest {
+ protected:
+  std::shared_ptr<ValueSegment<pmr_string>> vs_str = std::make_shared<ValueSegment<pmr_string>>();
+};
```

Comment:
I am not sure about the Test lifecycle here - is this object cleared after each test? If you are not sure, it might be good to add some logic for teardown and setup, where the object is constructed and cleared. That might ensure removing side effects.

Reply:
We copied this code from tests for `FixedStrignDictionary`. As `StorageVariableStringDictionarySegmentTest` is (through some levels of inheritance) a test fixture, [gtest uses a fresh instance for each test](http://google.github.io/googletest/faq.html#CtorVsSetUp:~:text=The%20first%20thing%20to%20remember%20is%20that%20GoogleTest%20does%20not%20reuse%20the%20same%20test%20fixture%20object%20across%20multiple%20tests.).